app/src/main/java/com/graeberj/taskman/auth/domain/repository/AuthRepositoryImpl.kt

```diff
@@ -30,43 +40,42 @@ class AuthRepositoryImpl(private val api: ApiService) : AuthRepository {
         password: String,
         fullName: String
     ): Resource<Boolean> {
-        val requestBody =
-            RegistrationRequestDto(email = email, fullName = fullName, password = password)
-        return try {
+        val requestBody = RegistrationRequestDto(email = email, fullName = fullName, password = password)
+
+        val result = authenticatedRetrofitCall(serializer) {
             api.registerUser(requestBody)
-            Resource.Success(data = true)
-        } catch (e: Exception) {
-            if (e == CancellationException()){
-                throw e
-            } else {
-                Resource.Error(message = e.toString())
-            }
+            AuthResult.Authorized(true)
+        }
+        return when (result) {
+            is AuthResult.Authorized -> Resource.Success(data = true)
+            is AuthResult.Unauthorized -> Resource.Error(UiText.DynamicString("Unauthorized access"))
+            is AuthResult.Error -> Resource.Error(UiText.DynamicString(result.message.toString()))
         }
     }
 
+
     override suspend fun authenticateUser(): Resource<Boolean> {
-        return try {
+        val result = authenticatedRetrofitCall(serializer) {
             api.checkAuthentication()
-            Resource.Success(data = true)
-        } catch (e: Exception) {
-            if (e == CancellationException()){
-                throw e
-            } else {
-                Resource.Error(message = e.toString())
-            }
+            AuthResult.Authorized(true) // Authentication success returns a boolean.
```

Comment:
Returning a boolean here seems redundant, since in which scenario would you return `AuthResult.Authorized(false)`? The fact that you return Authorized already contains the info that everything went well. You can therefore safely return `AuthResult.Authorized(Unit)`